### PR TITLE
Adds basic support for chronological log group style cert validation

### PR DESCRIPTION
**Pull request by AlCutter**
As defined here: https://github.com/GoogleChrome/ct-policy/issues

### CODE REVIEW

File: trillian/ctfe/configpb/config.proto [hunk 1]
```diff
@@ -30,4 +31,13 @@ message LogConfig {
   keyspb.PublicKey public_key = 5;
   bool reject_expired = 6;
   repeated string ext_key_usages = 7;
+	// not_after_start defines the start of the range of acceptable NotAfter
+	// values, inclusive.
+  google.protobuf.Timestamp not_after_start = 8;
+	// not_after_limit defines the end of the range of acceptable NotAfter values,
+	// exlusive.
```
Comment (daviddrysdale) [inline trillian/ctfe/configpb/config.proto]: exclusive

File: trillian/ctfe/configpb/config.proto [hunk 2]
```diff
@@ -30,4 +31,13 @@ message LogConfig {
   keyspb.PublicKey public_key = 5;
   bool reject_expired = 6;
   repeated string ext_key_usages = 7;
+	// not_after_start defines the start of the range of acceptable NotAfter
+	// values, inclusive.
```
Comment (daviddrysdale) [inline trillian/ctfe/configpb/config.proto]: Worth explicitly mentioning behaviour for zero value (and below)?

File: trillian/ctfe/handlers.go [hunk 3]
```diff
@@ -183,6 +185,13 @@ type CertValidationOpts struct {
 	trustedRoots *PEMCertPool
 	// rejectExpired indicates whether certificate validity period should be used during chain verification
 	rejectExpired bool
+	// notAfterStart is the earliest notAfter date which will be accepted.
```
Comment (daviddrysdale) [inline trillian/ctfe/handlers.go]: Does a pointer value of nil or a time.Time of zero mean no validation? Or both?

File: trillian/ctfe/cert_checker_test.go [hunk 4]
```diff
@@ -153,6 +153,130 @@ func TestValidateChain(t *testing.T) {
...
+	validateOpts := CertValidationOpts{
+		trustedRoots: fakeCARoots,
+		extKeyUsages: []x509.ExtKeyUsage{x509.ExtKeyUsageAny},
+	}
+	chain := pemsToDERChain(t, []string{testonly.LeafSignedByFakeIntermediateCertPEM, testonly.FakeIntermediateCertPEM})
+	leaf, err := x509.ParseCertificate(chain[0])
+	if err != nil {
+		t.Fatalf("Failed to parse golden certificate DER: %v", err)
+	}
+	t.Logf("Cert expiry date: %v", leaf.NotAfter)
```
Comment (daviddrysdale) [inline trillian/ctfe/cert_checker_test.go]: Expiry date looks irrelevant here -- left over copypasta?

File: trillian/ctfe/configpb/config.proto (hunk 2 above)
Comment (AlCutter) [inline trillian/ctfe/configpb/config.proto]: yup - done.

File: trillian/ctfe/configpb/config.proto (hunk 1 above)
Comment (AlCutter) [inline trillian/ctfe/configpb/config.proto]: done

File: trillian/ctfe/handlers.go (hunk 3 above)
Comment (AlCutter) [inline trillian/ctfe/handlers.go]: nil means no checking, Zero time would be a cut off of whatever time that works out to (epoch, presumably?)

File: trillian/ctfe/cert_checker_test.go (hunk 4 above)
Comment (AlCutter) [inline trillian/ctfe/cert_checker_test.go]: kinda - I was printing it just to I knew what it was when tests failed :)
Though that info is also part of the error msg, so I ding'd it.

**Comment by AlCutter** [comment]
PTAL

File: trillian/ctfe/configpb/config.proto [hunk 5]
```diff
@@ -30,4 +31,15 @@ message LogConfig {
...
+  // Leaving this unset implies no lower bound to the range.
+  google.protobuf.Timestamp not_after_start = 8;
+	// not_after_limit defines the end of the range of acceptable NotAfter values,
+	// exclusive.
+  // Leaving this unset implies no upper bound to the range.
```
Comment (daviddrysdale) [inline trillian/ctfe/configpb/config.proto]: indentation looks hinky in GitHub for the new fields -- tabs/spaces?

Comment (AlCutter) [inline trillian/ctfe/configpb/config.proto]: done
